### PR TITLE
Relax base upper bound for GHC 8.6.1

### PR DESCRIPTION
**Pull request by alanz**

### CODE REVIEW

File: HTTP.cabal [hunk 1]
```diff
@@ -109,7 +109,7 @@ Library
 
   -- note the test harness constraints should be kept in sync with these
   -- where dependencies are shared
-  Build-depends: base >= 4.3.0.0 && < 4.12, parsec >= 2.0 && < 3.2
+  Build-depends: base >= 4.3.0.0 && < 4.14, parsec >= 2.0 && < 3.2
```
[the remark this answers is not in the document]
Comment (alanz) [reply HTTP.cabal]: I guess. But the next major release will be even, so this will work. I can change it though.

**Comment by hvr** [comment]
@ganesh After convincing myself that `HTTP`'s use of `base` isn't affected by the changes in `base-4.12`, nor that any suspicious new compile warnings were emitted, and also verifying that the testsuite still passed, I've performed the revision at

http://hackage.haskell.org/package/HTTP-4000.3.12/revisions/

to relax from `base < 4.12` to `base < 4.13` as this was causing not-easy-to-workaround problems for users of the recent final GHC 8.6.1 release when trying to bootstrap `cabal-install-2.4.0.0` from scratch.

Since a revision sufficed to address this, there's nothing to be done (i.e. in particular there's no need for uploading a (redundant) release to Hackage merely for this metadata change) other than update the in-Git .cabal file to reflect this for future releases.

File: HTTP.cabal (hunk 1 above)
[the remark this answers is not in the document]
Comment (hsenag) [reply HTTP.cabal]: So base is moving to an odd/even scheme after this version?

(Sorry for taking a while on this, but there was no rush after hvr's bump on hackage)

**Comment by hsenag** [comment]
Superceded by a80d6c9720ea1a7c31bcf10b262deb6c83197341

**Pull request closed without merging**